File: src/mongodb/collections/AuctionTags.cs
using System.Runtime.Serialization;
using Cyotek.Data.Nbt;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

public class AuctionTags
{
    [BsonId]
    [DataMember]
    public ObjectId _id { get; set; }
    /// <summary>
    /// The name of the tag.
    /// </summary>
    public string Name;
    /// <summary>
    /// The type of tag.
    /// </summary>
    public TagType Type = TagType.None;
    /// <summary>
    /// Is a list of "Name" that point to it's tag. Used for compound tags
    /// </summary>
    public List<string> Tags = new List<string>();
    /// <summary>
    /// A list of values this tag has been seen with.
    /// </summary>
    public List<string> Values = new List<string>();

    public AuctionTags(string _Name, TagType _Type, string _Value, List<string>? _Tags = null)
    {
        Name = _Name;
        Type = _Type;
        Values.Add(_Value);

        if (_Tags != null)
            Tags = _Tags;
    }
}

File: apps/discord-bot/app/src/main/java/discordbot/mongodb/collections/AuctionTags.java
package discordbot.mongodb.collections;

import java.util.ArrayList;

import org.bson.BsonType;
import org.bson.codecs.pojo.annotations.BsonId;
import org.bson.codecs.pojo.annotations.BsonRepresentation;
import org.bson.types.ObjectId;

import net.querz.nbt.tag.Tag;

public class AuctionTags {
    @BsonId
    public ObjectId _id;
    /**
     * The name of the tag.
     */
    public String Name;
    /**
     * The type for the tag.
     */
    @BsonRepresentation(BsonType.STRING)
    public Tag<?> Type;
    /**
     * A list of values this tag has been seen with.
     */
    public ArrayList<String> Values = new ArrayList<String>();

    public AuctionTags(String _Name, Tag<?> _Type) {
        Name = _Name;
        Type = _Type;
    }
}
